include/boost/geometry/strategies/transform/matrix_transformers.hpp

```diff
@@ -41,6 +44,102 @@ namespace boost { namespace geometry
 namespace strategy { namespace transform
 {
 
+namespace detail { namespace matrix_transformer
+{
+
+template
+<
+    typename CalculationType,
+    typename P2,
+    std::size_t Dimension2,
+    std::size_t Dimension2Count
+>
+struct set_point_from_vec
+{
+    typedef CalculationType ct;
+    static inline void apply(P2& p2, qvm::vec<ct, Dimension2 + 1>& p2temp)
+    {
+        typedef typename geometry::coordinate_type<P2>::type ct2;
+        set<Dimension2Count>(p2, boost::numeric_cast<ct2>(qvm::A<Dimension2Count>(p2temp)));
+        set_point_from_vec<CalculationType, P2, Dimension2, Dimension2Count-1>::apply(p2, p2temp);
```

Comment:
Typically we're counting from `0` to `Dimension2Count` not the other way around. Also then `-1`  is not needed.

Reply:
Thanks, I will change the order of iteration.